LinuxTools builders fails to (mabye?) push at the end of PR builds

Discovered through https://github.com/rust-lang/rust/pull/62766 the end of a build failed with:

```
2019-07-18T17:43:29.7626178Z Verifying status of book...
2019-07-18T17:43:29.7645772Z Verifying status of nomicon...
2019-07-18T17:43:29.7656763Z Verifying status of reference...
2019-07-18T17:43:29.7669719Z Verifying status of rust-by-example...
2019-07-18T17:43:29.7681535Z Verifying status of edition-guide...
2019-07-18T17:43:29.7694011Z Verifying status of rls...
2019-07-18T17:43:29.7707209Z Verifying status of rustfmt...
2019-07-18T17:43:29.7718929Z Verifying status of clippy-driver...
2019-07-18T17:43:29.7730859Z Verifying status of miri...
2019-07-18T17:43:29.7744226Z Verifying status of embedded-book...
2019-07-18T17:43:29.7755764Z Verifying status of rustc-guide...
2019-07-18T17:43:29.7857440Z Cloning into 'rust-toolstate'...
2019-07-18T17:43:30.5204588Z The state of "rustc-guide" has changed from "test-pass" to ""
2019-07-18T17:43:30.5441945Z [master d757389] (linux CI update)
2019-07-18T17:43:30.5442055Z  1 file changed, 1 insertion(+)
2019-07-18T17:43:31.2635801Z remote: Invalid username or password.
2019-07-18T17:43:31.2636960Z fatal: Authentication failed for 'https://github.com/rust-lang-nursery/rust-toolstate.git/'
2019-07-18T17:43:31.5986192Z From https://github.com/rust-lang-nursery/rust-toolstate
2019-07-18T17:43:31.5986526Z  * branch            master     -> FETCH_HEAD
2019-07-18T17:43:31.6172922Z HEAD is now at 601d596 Merge pull request #13 from atouchet/links
2019-07-18T17:43:31.6293057Z The state of "rustc-guide" has changed from "test-pass" to ""
2019-07-18T17:43:31.6464398Z [master 8d4ee6f] (linux CI update)
2019-07-18T17:43:31.6464464Z  1 file changed, 1 insertion(+)
2019-07-18T17:43:31.9791628Z fatal: could not read Username for 'https://github.com': No such device or address
2019-07-18T17:43:32.3359605Z From https://github.com/rust-lang-nursery/rust-toolstate
2019-07-18T17:43:32.3360019Z  * branch            master     -> FETCH_HEAD
2019-07-18T17:43:32.3502731Z HEAD is now at 601d596 Merge pull request #13 from atouchet/links
2019-07-18T17:43:32.3618961Z The state of "rustc-guide" has changed from "test-pass" to ""
2019-07-18T17:43:32.3803151Z [master 55dd91f] (linux CI update)
2019-07-18T17:43:32.3803223Z  1 file changed, 1 insertion(+)
2019-07-18T17:43:32.6895480Z fatal: could not read Username for 'https://github.com': No such device or address
2019-07-18T17:43:34.0220883Z From https://github.com/rust-lang-nursery/rust-toolstate
2019-07-18T17:43:34.0221978Z  * branch            master     -> FETCH_HEAD
2019-07-18T17:43:34.0372608Z HEAD is now at 601d596 Merge pull request #13 from atouchet/links
2019-07-18T17:43:34.0478844Z The state of "rustc-guide" has changed from "test-pass" to ""
2019-07-18T17:43:34.0653889Z [master 69cc905] (linux CI update)
2019-07-18T17:43:34.0654492Z  1 file changed, 1 insertion(+)
2019-07-18T17:43:34.3745720Z fatal: could not read Username for 'https://github.com': No such device or address
2019-07-18T17:43:34.7285854Z From https://github.com/rust-lang-nursery/rust-toolstate
2019-07-18T17:43:34.7286707Z  * branch            master     -> FETCH_HEAD
2019-07-18T17:43:34.7437374Z HEAD is now at 601d596 Merge pull request #13 from atouchet/links
2019-07-18T17:43:34.7553136Z The state of "rustc-guide" has changed from "test-pass" to ""
2019-07-18T17:43:34.7732285Z [master 69cc905] (linux CI update)
2019-07-18T17:43:34.7732378Z  1 file changed, 1 insertion(+)
2019-07-18T17:43:35.0867854Z fatal: could not read Username for 'https://github.com': No such device or address
2019-07-18T17:43:37.4349692Z From https://github.com/rust-lang-nursery/rust-toolstate
2019-07-18T17:43:37.4350364Z  * branch            master     -> FETCH_HEAD
2019-07-18T17:43:37.4489321Z HEAD is now at 601d596 Merge pull request #13 from atouchet/links
2019-07-18T17:43:38.0902365Z ##[error]Bash exited with code '1'.
```

That could be a weird unauthenticated pull from GitHub but I suspect that it's a push which will inevitably fail. I think we haven't configured the toolstate builder to not do anything on PRs on push?

--- a/src/ci/docker/x86_64-gnu-tools/checktools.sh
+++ b/src/ci/docker/x86_64-gnu-tools/checktools.sh
@@ -112,7 +112,7 @@ $COMMIT\t$(cat "$TOOLSTATE_FILE")
 }
 
 if [ "$RUST_RELEASE_CHANNEL" = nightly ]; then
-    if [ -n "${TOOLSTATE_REPO_ACCESS_TOKEN+is_set}" ]; then
+    if [ -n "${TOOLSTATE_PUBLISH+is_set}" ]; then
         . "$(dirname $0)/repo.sh"
         MESSAGE_FILE=$(mktemp -t msg.XXXXXX)
         echo "($OS CI update)" > "$MESSAGE_FILE"

